packages/account/src/Sections/Profile/PhoneVerification/__test__/resend-code-timer.spec.tsx

```diff
@@ -46,22 +78,54 @@ describe('ConfirmPhoneNumber', () => {
         expect(resend_button_after).toBeInTheDocument();
     });
 
+    it('should trigger setShouldShowDidntGetTheCodeModal when Didn`t get the code is clicked', () => {
+        const setShouldShowDidntGetTheCodeModal = jest.fn();
+        render(
+            <ResendCodeTimer
+                resend_code_text='Didn’t get the code?'
+                count_from={6}
+                setStartTimer={jest.fn()}
+                start_timer={false}
+                setShouldShowDidntGetTheCodeModal={setShouldShowDidntGetTheCodeModal}
+            />
+        );
+        const resend_button_after = screen.getByRole('button', { name: 'Didn’t get the code?' });
+        userEvent.click(resend_button_after);
+        expect(setShouldShowDidntGetTheCodeModal).toHaveBeenCalled();
+    });
+
     it('should check if title displays countdown time when timer starts and value of resend_code_text is Didn’t get the code?', async () => {
-        render(<ResendCodeTimer resend_code_text='Didn’t get the code?' count_from={6} />);
-        const resend_button = screen.getByRole('button', { name: 'Didn’t get the code?(6s)' });
+        render(
+            <ResendCodeTimer
+                resend_code_text='Didn’t get the code?'
+                count_from={6}
+                setStartTimer={jest.fn()}
+                start_timer={true}
+                setShouldShowDidntGetTheCodeModal={jest.fn()}
+            />
+        );
+        const resend_button = screen.getByRole('button', { name: 'Didn’t get the code? (6s)' });
         userEvent.click(resend_button);
 
         await waitFor(
             () => {
-                const resend_button = screen.getByRole('button', { name: 'Didn’t get the code?(4s)' });
+                const resend_button = screen.getByRole('button', { name: 'Didn’t get the code? (4s)' });
                 expect(resend_button).toBeInTheDocument();
             },
             { timeout: 4000 }
```

Comment:
can we try to use jest fake timers here and not make out tests working for so long time :)